x/payments/spec/03_messages.md

```diff
@@ -0,0 +1,145 @@
+# Messages
+
+In this section we describe the processing of the payment messages and the corresponding updates to the state. All created/modified state objects specified by each message are defined within the [state](./02_state.md) section.
+
+## MsgCreatePaymentTemplate
+
+PaymentTemplate can be created by any address using `MsgCreatePaymentTemplate`.
+
+| **Field**              | **Type**           | **Description**                                                                                               |
+|:-----------------------|:-----------------  |:--------------------------------------------------------------------------------------------------------------|
+| CreatorDid             | `did.DID`          | 
+| PaymentTemplate        | `PaymentTemplate`  | 
+
+```go
+type MsgCreatePaymentTemplate struct {
+	CreatorDid      did.Did         `json:"creator_did" yaml:"creator_did"`
+	PaymentTemplate PaymentTemplate `json:"payment_template" yaml:"payment_template"`
+}
+```
+
+This message creates and stores the `MsgCreatePaymentTemplate` object at appropriate indexes. 
+
+## MsgCreatePaymentContract 
+
+The owner of a bond can edit some of the payment's parameters using `MsgCreatePaymentContract`.
+
+| **Field**              | **Type**           | **Description**                                                                                               |
+|:-----------------------|:-------------------|:--------------------------------------------------------------------------------------------------------------|
+| CreatorDid             | `did.Did`          | Did of the user
+| PaymentTemplateId      | `string`           | ID of the paymentTemplate
+| PaymentContractId      | `string`           | ID of the PaymentContract
+| Payer                  | `sdk.AccAddress`   | Address of the payer
+| CanDeauthorise         | `bool`             | Bool of de_authorise
+| DiscountId             | ` sdk.Uint`        | Any discount given
+
+```go
+type MsgCreatePaymentContract struct {
+	CreatorDid        did.Did        `json:"creator_did" yaml:"creator_did"`
+	PaymentTemplateId string         `json:"payment_template_id" yaml:"payment_template_id"`
+	PaymentContractId string         `json:"payment_contract_id" yaml:"payment_contract_id"`
+	Payer             sdk.AccAddress `json:"payer" yaml:"payer"`
+	CanDeauthorise    bool           `json:"can_deauthorise" yaml:"can_deauthorise"`
+	DiscountId        sdk.Uint       `json:"discount_id" yaml:"discount_id"`
+}
+```
+This message is expected to fail if:
+- any editable field violates the restrictions set for the same field in `MsgCreatePaymentContract`
+- all editable fields are `"[do-not-modify]"`
+
+## MsgCreateSubscription 
+
+The owner of a bond can edit some of the bond's parameters using `MsgCreateSubscription`.
```

Comment:
This mentions bonds but we're not dealing with bonds here.

Reply:
removed